lib/phoenix_live_view/async.ex

```diff
@@ -3,13 +3,91 @@ defmodule Phoenix.LiveView.Async do
 
   alias Phoenix.LiveView.{AsyncResult, Socket, Channel}
 
-  def start_async(%Socket{} = socket, key, func, opts \\ []) when is_function(func, 0) do
-    run_async_task(socket, key, func, :start, opts)
+  @warn_on_expensive_assigns Application.compile_env(
+                               :phoenix_live_view,
+                               :warn_on_expensive_assigns,
+                               false
+                             )
+
+  if @warn_on_expensive_assigns do
+    def start_async(%Socket{} = socket, key, func, opts \\ []) when is_function(func, 0) do
```

Comment:
Instead of wrapping the whole start_async, you can create a private function called validate_function_env that is a no-op if the flag is false.

Reply:
I'm not sure about showing the warning in both cases. It could be confusing to see it twice. As we don't run the function in the disconnected render anyways I think it's reasonable to only show it in the connected case.

Reply:
Agreed. thank you!